File: VB/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraReports.UI
' ...

Namespace SuppressBlankLines
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrLabel5_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrLabel5.BeforePrint
			' Get the current value of the "Fax" data column
			Dim faxName As String = GetCurrentColumnValue("Fax").ToString()

			' Check if the value is null, then cancel the printing
			'if (faxName == "") e.Cancel = true;
			If faxName = "" Then
				CType(sender, XRLabel).Text = String.Empty
			Else
				CType(sender, XRLabel).Text = "Fax"
			End If
		End Sub

		Private Sub xrLabel6_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrLabel6.BeforePrint
			' Get the current value of the "Fax" data column
			Dim faxName As String = GetCurrentColumnValue("Fax").ToString()

			' Check if the value is null, then cancel the printing
			'if (faxName == "") e.Cancel = true;
			If faxName = "" Then
				CType(sender, XRLabel).Text = String.Empty
			End If
		End Sub

	End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using DevExpress.XtraReports.UI;
// ...

namespace SuppressBlankLines {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void xrLabel5_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            // Get the current value of the "Fax" data column
            string faxName = GetCurrentColumnValue("Fax").ToString();

            // Check if the value is null, then cancel the printing
            //if (faxName == "") e.Cancel = true;
            if(faxName == "")
                ((XRLabel)sender).Text = String.Empty;
            else
                ((XRLabel)sender).Text = "Fax";
        }

        private void xrLabel6_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            // Get the current value of the "Fax" data column
            string faxName = GetCurrentColumnValue("Fax").ToString();

            // Check if the value is null, then cancel the printing
            //if (faxName == "") e.Cancel = true;
            if(faxName == "")
                ((XRLabel)sender).Text = String.Empty;
        }

    }
}
